code/modules/surgery/organs/blood.dm

```diff
@@ -1,87 +1,40 @@
 /****************************************************
 				BLOOD SYSTEM
 ****************************************************/
-
-/mob/living/carbon/human/var/datum/reagents/vessel	//Container for blood and BLOOD ONLY. Do not transfer other chems here.
-
-//Initializes blood vessels
-/mob/living/carbon/human/proc/make_blood()
-
-	if(vessel)
+/mob/living/carbon/human/proc/suppress_bloodloss(amount)
+	if(bleedsuppress)
 		return
+	else
+		bleedsuppress = 1
+		spawn(amount)
+			bleedsuppress = 0
+			if(stat != DEAD && bleed_rate)
+				src << "<span class='warning'>The blood soaks through your bandage.</span>"
 
-	vessel = new/datum/reagents(max_blood)
-	vessel.my_atom = src
 
-	if(species && species.exotic_blood)
-		vessel.add_reagent(species.exotic_blood, max_blood)
-	else
-		vessel.add_reagent("blood", max_blood)
-		for(var/datum/reagent/blood/B in vessel.reagent_list)
-			if(B.id == "blood")
-				B.data = list(
-				"donor" = src,
-				"viruses" = null,
-				"blood_DNA" = dna.unique_enzymes,
-				"blood_colour" = species.blood_color,
-				"blood_type" = b_type,
-				"resistances" = null,
-				"trace_chem" = null)
-	spawn(1)
-		fixblood()
-
-//Resets blood data
-/mob/living/carbon/human/proc/fixblood()
-	for(var/datum/reagent/blood/B in vessel.reagent_list)
-		if(B.id == "blood")
-			B.data = list(
-			"donor" = src,
-			"viruses" = null,
-			"blood_DNA" = dna.unique_enzymes,
-			"blood_colour" = species.blood_color,
-			"blood_type" = b_type,
-			"resistances" = null,
-			"trace_chem" = null)
 
 // Takes care blood loss and regeneration
 /mob/living/carbon/human/proc/handle_blood()
-	var/blood_volume
+	var/list/blood_data = get_blood_data(get_blood_id())//PROCCEPTION
 	if(species && species.flags & NO_BLOOD)
+		bleed_rate = 0
 		return
+
 	if(stat != DEAD && bodytemperature >= 170)	//Dead or cryosleep people do not pump the blood.
 		if(species.exotic_blood)
```

Comment:
this species exotic blood thing isn't even needed anymore, with this system, I don't think.